test/e2e/e2e_test.go

```diff
@@ -863,3 +866,96 @@ var _ = Describe("e2e multiple external gateway update validation", func() {
 		}
 	})
 })
+
+func checkPodSleep(f *framework.Framework, nodeName, podName string, errChan chan error) {
+	contName := fmt.Sprintf("%s-container", podName)
+
+	command := []string{
+		"bash", "-c",
+		"sleep 10",
+	}
+
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: podName,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:    contName,
+					Image:   framework.AgnHostImage,
+					Command: command,
+				},
+			},
+			NodeName:      nodeName,
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+	podClient := f.ClientSet.CoreV1().Pods(f.Namespace.Name)
+	result, err := podClient.Create(pod)
+	if err != nil {
+		errChan <- err
+		return
+	}
+
+	framework.Logf("Created pod %q with IP %q %q", result.GetName(), result.Status.PodIPs, result.Status.PodIPs)
+
+	err = e2epod.WaitForPodNotPending(f.ClientSet, f.Namespace.Name, podName)
+	if err != nil {
+		errChan <- err
+		return
+	}
+
+	err = e2epod.WaitForPodSuccessInNamespace(f.ClientSet, podName, f.Namespace.Name)
+
+	if err != nil {
+		logs, logErr := e2epod.GetPodLogs(f.ClientSet, f.Namespace.Name, pod.Name, contName)
```

Comment:
is it possible this never returns?

Reply:
It's definitely possible but while it's waiting it continuously prints to the log, and it has its own timeout of 5 minutes: https://github.com/kubernetes/kubernetes/blob/v1.17.2/test/e2e/framework/pod/wait.go#L375